src/Microsoft.Management.UI.Internal/ManagementList/FilterCore/FilterRules/FilterRuleExtensions.cs

```diff
@@ -27,7 +27,34 @@ public static class FilterRuleExtensions
         /// </returns>
         public static FilterRule DeepCopy(this FilterRule rule)
         {
-            return rule.Clone();
+            ArgumentNullException.ThrowIfNull(rule);
+
+#pragma warning disable SYSLIB0050
+            Debug.Assert(rule.GetType().IsSerializable, "rule is serializable");
+
+#pragma warning disable SYSLIB0011
+#pragma warning disable SYSLIB0050
```

Comment:
Those warnings are repeatedly disabled, which seems to be unnecessary. Can you maybe just disable both warnings for this whole method and restore them at the end of the method?

Reply:
You're right, new commit should do this optimally